### DISCUSSION TITLE
After saving, the custom traits do not appear.

### DISCUSSION THREAD

**Discussion by capoia** [GrapesJS/grapesjs#5481] [Q&A] [answered]
Hello everything is fine? I created a plugin for grapes that has a form, this form has special traits (where it will redirect to etc.), when I drag the form and go to configs it shows the traits, however when saving in the bank (saved in json, I will leave an example is attached) and refreshing the page renders the form correctly, but the custom traits no longer appear.

In the component definition:
```js
  defaults: {
    tagName: "form",
    droppable: false,
    attributes: { method: "POST" },
    traits: [
      {
        label: "Api key",
        type: "text",
        name: "keycubo",
        changeProp: true,
        required: true,
      },
      {
        label: "Funíl",
        type: "select",
        name: "pipe",
        changeProp: true,
        options: pipes,
        required: true,
      },
      {
        label: "Redirecionar para",
        type: "text",
        name: "redirect",
        changeProp: true,
      },
      {
        label: "Adicionar na fila",
        type: "select",
        name: "queue",
        changeProp: true,
        options: [
          { value: "true", name: "Sim" },
          { value: "false", name: "Não" },
        ],
        default: "false",
        required: true,
      },
    ],
    ...
  }
```

In the block definition:
```js
{
    label: "Formulário Criação Cubo",
    content: { type: "cubosuite-form" },
    category: "Formulários",
    ...
}
```

video of the problem:
https://github.com/GrapesJS/grapesjs/assets/21159928/fa5b0b46-6e85-4a9e-9b36-7cafcb1af1e6

part of the json (components) that I save and then load into the project:
https://api.npoint.io/c2c5e33ffcffae58434c/0/type

**Comment by mosh-tudor** [comment, marked as answer]
@capoia What do you have in "isComponent" part?

You should have something like this:
```js
    isComponent(el) {
      if (el?.tagName === "form" && el?.classList?.contains('cubosuite-form')) {
        return { type: "cubosuite-form" };
      }
    },
```

**Reply by capoia** [reply]
You're right, I hadn't done this part, as there isn't another form component added I didn't think I needed it, thank you very much.

**Comment by capoia** [comment]
@mosh-tudor In reality it seemed like it had worked, but it didn't, for some reason sometimes it recognizes the component and other times it gives a warning on the console saying that the component was not found (as you can see in the video).

https://github.com/GrapesJS/grapesjs/assets/21159928/61958714-c2b8-4dd5-ad0c-47d33c9a27db

I put a console.log in the isComponent function, and I notice that every time I get an error it is not executed.
Something that could make this function not be called?

Below is a print of the construction of the component/blocks
![CleanShot 2023-10-25 at 16 57 34](https://github.com/GrapesJS/grapesjs/assets/21159928/ee738720-faa4-45c9-acb4-6a9a523bb10
![CleanShot 2023-10-25 at 16 58 05](https://github.com/GrapesJS/grapesjs/assets/21159928/5e5d5062-b440-4527-b61b-75186e5b7079)
e)
![CleanShot 2023-10-25 at 16 58 33](https://github.com/GrapesJS/grapesjs/assets/21159928/0053163e-418d-428a-905a-408fba77d6a9)
